### PR TITLE
Tests improvements

### PR DESCRIPTION
**Pull request by MiroslavGatsanoga**

### CODE REVIEW

File: Dockerfile.tests [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM golang:1.10
+
```
Comment (epavlova) [inline Dockerfile.tests]: I know that migrating to modules will be in different PR, but we can fix the go version here.

File: concepts/concepts_service_test.go [hunk 2]
```diff
@@ -78,776 +82,21 @@ var conceptsDriver ConceptService
...
-				YearFounded:            1951,
-				EmailAddress:           "info@strix.com",
-				LeiCode:                "213800KZEW5W6BZMNT62",
-				ParentOrganisation:     parentOrgUUID,
-			},
-		},
-	}
+func getAggregatedConcept(t *testing.T, name string) AggregatedConcept {
+	ac := AggregatedConcept{}
+	json.Unmarshal(helperLoadBytes(t, name), &ac)
```
Comment (epavlova) [inline concepts/concepts_service_test.go]: We can handle the error here.

File: concepts/concepts_service_test.go [hunk 3]
```diff
@@ -919,97 +168,10 @@ func getOrganisationWithAllCountries() AggregatedConcept {
...
-				TerminationDate:  anotherMembershipRole.TerminationDate,
-				MembershipRoles: []MembershipRole{
-					anotherMembershipRole,
-				},
-			},
-		},
-	}
+func getConcept(t *testing.T, name string) Concept {
+	c := Concept{}
+	json.Unmarshal(helperLoadBytes(t, name), &c)
```
Comment (epavlova) [inline concepts/concepts_service_test.go]: We can handle the error here.

File: .circleci/config.yml [hunk 4]
```diff
@@ -37,7 +37,7 @@ jobs:
       - run:
           name: Run tests and create coverage report
           command: |
-            go test ./... -race -cover -coverprofile=$CIRCLE_TEST_REPORTS/coverage.out | go-junit-report > $CIRCLE_TEST_REPORTS/golang/junit.xml
+            go test ./... -race -tags=integration -cover -coverprofile=$CIRCLE_TEST_REPORTS/coverage.out | go-junit-report > $CIRCLE_TEST_REPORTS/golang/junit.xml
```
[the remark this answers is not in the document]
Comment (MiroslavGatsanoga) [reply .circleci/config.yml]: Since this PR is about the tests only, I have updated only the Go version for the tests. Updating the Go version for the project e.g. in `.circleci/config.yml` and `Dockerfile`, will follow in the next PR that I am preparing to open (migration to modules will be done there as well).

File: concepts/handlers_test.go [hunk 5]
```diff
@@ -127,44 +152,47 @@ func TestGetHandler(t *testing.T) {
...
 		},
 		{
 			name: "ReadError",
-			req:  newRequest("GET", fmt.Sprintf("/dummies/%s", knownUUID)),
+			req:  newRequest("GET", fmt.Sprintf("/dummies/%s", knownUUID), t),
 			ds: &mockConceptService{
-				uuid:        knownUUID,
-				conceptType: "Dummy",
-				failRead:    true,
+				read: func(uuid string, transID string) (interface{}, bool, error) {
```
[the remark this answers is not in the document]
Comment (MiroslavGatsanoga) [reply concepts/handlers_test.go]: I am not sure if I understand this one correctly, but if it is regarding the naming of `uuid` both are fine as described [here](https://github.com/golang/go/wiki/CodeReviewComments#initialisms), so will leave it as it is. Please let me know if you meant something else here, so that I can address it.

File: concepts/handlers_test.go [hunk 6]
```diff
@@ -127,44 +152,47 @@ func TestGetHandler(t *testing.T) {
...
 			body:        errorMessage("TEST failing to READ"),
 		},
 		{
 			name: "BadConceptOrPath",
-			req:  newRequest("GET", fmt.Sprintf("/dummies/%s", knownUUID)),
+			req:  newRequest("GET", fmt.Sprintf("/dummies/%s", knownUUID), t),
 			ds: &mockConceptService{
-				uuid:        knownUUID,
-				conceptType: "not-dummy",
+				read: func(uuid string, transID string) (interface{}, bool, error) {
```
[the remark this answers is not in the document]
Comment (MiroslavGatsanoga) [reply concepts/handlers_test.go]: Same as[ this one](https://github.com/Financial-Times/concepts-rw-neo4j/pull/57#discussion_r346845563).

File: concepts/handlers_test.go [hunk 7]
```diff
@@ -208,10 +258,10 @@ func TestGtgHandler(t *testing.T) {
 	}
 }
 
-func newRequest(method, url string) *http.Request {
+func newRequest(method, url string, t *testing.T) *http.Request {
```
[the remark this answers is not in the document]
Comment (MiroslavGatsanoga) [reply concepts/handlers_test.go]: Same as[ this one](https://github.com/Financial-Times/concepts-rw-neo4j/pull/57#discussion_r346845563).

File: concepts/handlers_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (mchompalova) [reply concepts/handlers_test.go]: Thanks, this is what I meand and I learnt something new today :)
